actions/open-workflow-issue/action.yml

```diff
@@ -0,0 +1,76 @@
+name: Open workflow issue
+description: Open a github issue for a workflow failure
+# An issue is filed unless one for the same workflow is open already.
+# The issues are identitied by title which is built using the workflow
+# name.
+#
+# Required permissions:
+# * issues: write
+# * actions: read
+
+inputs:
+  comment_for_each_failure:
+    description: 'Should a comment be added to existing issue for every new failure'
+    required: false
+    default: false
+  fyi:
+    description: 'Text to add to the end of every comment; can be used to notify others'
+    required: false
+    default: ''
+
+runs:
+  using: "composite"
+  steps:
+    - name: Create issue
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      env:
+        COMMENT_FOR_EACH_FAILURE: ${{ inputs.comment_for_each_failure }}
+        FYI_STRING: ${{ inputs.fyi }}
+      with:
+        script: |
+          var path = require('path');
+
+          const reponame = context.repo.owner + "/" + context.repo.repo
+          const title = `[bug]: Workflow failure '${context.workflow}'`
+          const issues = await github.rest.search.issuesAndPullRequests({
+            q:  `${title}+in:title+label:bug+state:open+type:issue+repo:${reponame}`,
+          })
+          comment_for_each_failure = process.env.COMMENT_FOR_EACH_FAILURE == "true"
+
+          const run = await github.rest.actions.getWorkflowRun( {
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            run_id: context.runId,
+          })
+          const workflow_file = path.basename(run.data.path)
+
+          body = `### Workflow run failed for '${context.workflow}'.\n` +
+                 `Run: ${run.data.html_url}\n` +
+                 `Workflow: ${run.data.repository.html_url}/blob/${context.ref}/${run.data.path}\n` +
+                 `Workflow runs: ${run.data.repository.html_url}/actions/workflows/${workflow_file}\n` +
+                 `Trigger: ${context.eventName} on ${context.ref}\n` +
+                 `Date: ${run.data.run_started_at}` +
+                 process.env.FYI_STRING != `\n\nFYI: ` + process.env.FYI_STRING : ""
```

Comment:
This might be valid javascript (?) but it isn't obvious to me that it actually works. Does it?

Reply:
you're right, there is a typo in here (missing `?` operator)